pkg/controller/utils/datadog/metrics_forwarder_test.go

```diff
@@ -1029,3 +1029,49 @@ func Test_metricsForwarder_getSecretsFromCache(t *testing.T) {
 		})
 	}
 }
+
+func Test_getbaseURL(t *testing.T) {
+	type args struct {
+		dda *datadoghqv1alpha1.DatadogAgent
+	}
+	tests := []struct {
+		name string
+		args args
+		want string
+	}{
+		{
+			name: "Get default baseURL",
+			args: args{
+				dda: test.NewDefaultedDatadogAgent("foo", "bar", &test.NewDatadogAgentOptions{}),
+			},
+			want: "https://api.datadoghq.com",
+		},
+		{
+			name: "Compute baseURL from site when passing Site",
+			args: args{
+				dda: test.NewDefaultedDatadogAgent("foo", "bar", &test.NewDatadogAgentOptions{
+					Site: "datadoghq.eu",
+				}),
+			},
+			want: "https://api.datadoghq.eu",
+		},
+		{
+			name: "Compute baseURL from ddUrl when Site is not defined",
+			args: args{
+				dda: test.NewDefaultedDatadogAgent("foo", "bar", &test.NewDatadogAgentOptions{
+					NodeAgentConfig: &datadoghqv1alpha1.NodeAgentConfig{
+						DDUrl: datadoghqv1alpha1.NewStringPointer("https://test.url.com"),
+					}}),
+			},
+			want: "https://test.url.com",
+		},
+		// TODO: Add test cases.
```

Comment:
nit: can we add a test case where both `Site` and `DDUrl` are explicitly set to test the order of precedence?

Reply:
Done